From: Suryashankar Das <suryashankardas.2002@gmail.com>
Date: Wed, 10 Mar 2021 01:34:46 +0530
Subject: meta-seat-x11: fix event handling for x11 backend

This commit adds the events created in the function
`meta_seat_x11_notify_devices` to the clutter events queue, which
are currently only added to the stage queue making the events not
being picked up by the `clutter_seat_handle_event_post` function.
This results in devices not getting added to the devices-list of
`MetaInputSettings`.

Fixes the bug in which mouse and touchpad settings are not working in
the settings app during x11 session.
---
 src/backends/x11/meta-seat-x11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/x11/meta-seat-x11.c b/src/backends/x11/meta-seat-x11.c
index fc908ed..c419847 100644
--- a/src/backends/x11/meta-seat-x11.c
+++ b/src/backends/x11/meta-seat-x11.c
@@ -1325,7 +1325,7 @@ meta_seat_x11_notify_devices (MetaSeatX11  *seat_x11,
       event = clutter_event_new (CLUTTER_DEVICE_ADDED);
       clutter_event_set_device (event, device);
       clutter_event_set_stage (event, stage);
-      clutter_do_event (event);
+      _clutter_event_push (g_steal_pointer (&event), FALSE);
     }
 }
 
